api/models/student.py

```diff
@@ -67,3 +67,7 @@ class Meta:
             models.Index(fields=("project_status",), name="student_pr_status_idx"),
             models.Index(fields=("situational_status",), name="student_si_status_idx"),
         ]
+
+    @property
+    def has_groups(self) -> bool:
+        return self.groups.count() > 0
```

Comment:
Maybe `self.groups.exists()` ?
https://dev.to/codereviewdoctor/why-queryset-exists-is-more-efficient-than-queryset-count-2f3h